src/Pulsar.Client/Common/DTO.fs

```diff
@@ -525,27 +519,21 @@ type ConsumerStats = {
     IncomingMsgs: int
 }
 
-type EncryptionKeys(key: string, value: byte [], metadatas: Dictionary<string, string>) =
+type EncryptionKeys(key: string, value: byte [], metadata: Dictionary<string, string>) =
     member val Key = key
     member val Value = value
-    member val Metadata = metadatas
-    
-    new(key: string, value: byte []) =
-        EncryptionKeys(key, value, Dictionary<string, string>())
+    member val Metadata = metadata
 
     static member internal ToProto(encKey: EncryptionKeys) =
         let result = pulsar.proto.EncryptionKeys(Key = encKey.Key, Value = encKey.Value)
-        if not (isNull encKey.Metadata) then
-            encKey.Metadata
-            |> Seq.cast<KeyValue>
-            |> Seq.iter result.Metadatas.Add
+        for (KeyValue(k, v)) in encKey.Metadata do
+             result.Metadatas.Add(KeyValue(Key = k, Value = v))
```

Comment:
I think we should leave the check for null
User can implement IMessageEncryptor where Metadata is null